nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/ml/textract/StartAwsTextractJob.java

```diff
@@ -0,0 +1,134 @@
+    }
+
+    @Override
+    protected AmazonTextractClient createClient(ProcessContext context, AWSCredentialsProvider credentialsProvider, ClientConfiguration config) {
+        return (AmazonTextractClient) AmazonTextractClient.builder()
+                .withRegion(context.getProperty(REGION).getValue())
+                .withCredentials(credentialsProvider)
+                .build();
+    }
+
+    @Override
+    protected AmazonWebServiceResult sendRequest(AmazonWebServiceRequest request, ProcessContext context) {
+        String typeOfTextract = context.getProperty(TYPE.getName()).getValue();
+        AmazonWebServiceResult result;
+        switch (typeOfTextract) {
+            case DOCUMENT_ANALYSIS :
+                result = getClient().startDocumentAnalysis((StartDocumentAnalysisRequest) request);
+                break;
+            case DOCUMENT_TEXT_DETECTION:
+                result = getClient().startDocumentTextDetection((StartDocumentTextDetectionRequest) request);
+                break;
+            case EXPENSE_ANALYSIS :
+                result = getClient().startExpenseAnalysis((StartExpenseAnalysisRequest) request);
+                break;
+            default: throw new UnsupportedOperationException("Unsupported textract type.");
+        }
+        return result;
+    }
+
+    @Override
+    protected Class<? extends AmazonWebServiceRequest> getAwsRequestClass(ProcessContext context) {
+        String typeOfTextract = context.getProperty(TYPE.getName()).getValue();
+        Class<? extends AmazonWebServiceRequest>  result;
+        switch (typeOfTextract) {
+            case DOCUMENT_ANALYSIS:
+                result = StartDocumentAnalysisRequest.class;
+                break;
+            case DOCUMENT_TEXT_DETECTION:
+                result = StartDocumentTextDetectionRequest.class;
+                break;
+            case EXPENSE_ANALYSIS:
+                result = StartExpenseAnalysisRequest.class;
+                break;
+            default: throw new UnsupportedOperationException("Unsupported textract type.");
+        }
+        return result;
+    }
+
+    @Override
+    protected String getAwsTaskId(ProcessContext context, AmazonWebServiceResult amazonWebServiceResult) {
+        String typeOfTextract = context.getProperty(TYPE.getName()).getValue();
+        String  result;
+        switch (typeOfTextract) {
+            case DOCUMENT_ANALYSIS:
+                result = ((StartDocumentAnalysisResult) amazonWebServiceResult).getJobId();
+                break;
+            case DOCUMENT_TEXT_DETECTION:
+                result = ((StartDocumentTextDetectionResult) amazonWebServiceResult).getJobId();
+                break;
+            case EXPENSE_ANALYSIS:
+                result = ((StartExpenseAnalysisResult) amazonWebServiceResult).getJobId();
+                break;
+            default: throw new UnsupportedOperationException("Unsupported textract type.");
```

Comment:
The `typeOfTextract` string should be included:
```suggestion
            default: throw new UnsupportedOperationException("Unsupported textract type: " + typeOfTextract);
```

Reply:
Fixed, thx.